roles/mongodb/tasks/mongodb_auth.yml

```diff
@@ -0,0 +1,68 @@
+---
+- name: Ensure pymongo is installed for the mongodb_user ansible module
+  become: yes
+  pip:
+    name: pymongo
+
+- name: See if authorization is enabled in mongod.conf
+  check_mode: yes
+  lineinfile:
+    path: /etc/mongod.conf
+    insertafter: 'security:'
+    line: '  authorization: enabled'
```

Comment:
I think we'll need an additional `regexp` to make the `lineinfile` less error-prone.

Reply:
Isn't it sad that there's no yaml module? Ansible speaks yaml, but it can't easily change configuration vars in yaml :(
BTW, this was fun to figure out how to check if authorization is enabled via a check_mode task.

An additional regexp? To deal with alternative indentation? By default the file uses 2-space indentation. How often do people switch away from that default?

Reply:
There's a yaml editing module, but I hesitate to rely on it:
https://github.com/kwoodson/yedit

- build system feels weird: They are somehow squashing several files to generate the module, but they distribute all of the pieces with the finished module
- packaging feels weird: They're distributing the module inside of a role which is very non-standard
- distribution is difficult: It's not available on ansible-galaxy, so is it really ready for third party use? Do I really want to maintain the distribution of someone else's module for use in my playbook when they won't even bother to publish it on galaxy?

Reply:
Other approaches:
 - [this one](https://galaxy.ansible.com/greendayonfire/mongodb/) uses a template to create a temporary (templated) mongod.conf with authentication disabled while adding the admin and other users, start mongo with that config, then rebuild the real config with another template.
- [this one](https://galaxy.ansible.com/Stouts/mongodb/) uses a template for the config file that is an ini looking thing instead of yaml.
- others I looked at use templates for the conf file.

I don't want to use a template unless we're willing to control the whole conf file.

Reply:
I still think adding a `regexp` here will make it less error prone. Like not perfect, but much better and good enough without going into overcomplication.

Consider the following examples: 
```
  authorization: enabled 
   authorization: enabled
  authorization: enabled
  authorization:enabled
  authorization: "enabled"
  authorization: 'enabled'
  authorization : enabled
```

Reply:
Ah. I see. Makes sense. I add a regex then.

Reply:
https://github.com/StackStorm/ansible-st2/pull/162

Reply:
Ah. I see. I will add the regex then :)